main/src/library/Array.flix

```diff
@@ -57,10 +57,1390 @@ namespace Array {
     ///
     /// Helper for toList.
     ///
-    def toListHelper(a: Array[a], i: Int): List[a] & Impure =
+    def toListHelper(a: Array[a], i: Int32): List[a] & Impure =
         if (i == length(a))
             Nil
         else
             a[i] :: toListHelper(a, i + 1)
 
-}
+    ///
+    /// Returns `Some(x)` if `x` is the first element of `a`.
+    ///
+    /// Returns `None` if `a` is empty.
+    ///
+    pub def head(a: Array[a]): Option[a] & Pure =
+        if (length(a) > 0) Some(a[0] as & Pure) else None
+
+    ///
+    /// Returns `Some(x)` if `x` is the last element of `a`.
+    ///
+    /// Returns `None` if `a` is empty.
+    ///
+    pub def last(a: Array[a]): Option[a] & Pure =
+        let len = length(a);
+        if (len > 0) Some(a[len-1] as & Pure) else None
```

Comment:
In Julia they made an interesting design choice to add a special keyword for "the last element of an array":

https://en.wikibooks.org/wiki/Introducing_Julia/Arrays_and_tuples#Accessing_the_contents_of_arrays

Do you think that would be useful for us? or unneccessary?

I suspect they may have added it because their language is 1-indexed rather than 0-indexed (!)